app/controllers/auth.js

```diff
@@ -1,45 +1,64 @@
-var express  = require('express');
-var router   = express.Router();
-var mongoose = require('mongoose');
+// var express  = require('express');
+// var router   = express.Router();
+// var mongoose = require('mongoose');
 var User     = require('../models/user');
 var passport = require('passport');
 
-module.exports = function (app) {
-  app.use('/', router);
-};
+// module.exports = function (app) {
+//   app.use('/', router);
+// };
 
-router.get('/signup', function (req, res, next) {
-  res.render('auth/signup.ejs', { message: req.flash('signupMessage') });
-});
+// GET /signup
+function getSignup(req, res, next) {
+  res.render('auth/signup.ejs', {message: req.flash('signupMessage') });
+};
 
 // SIGN-UP: Create a new User
-router.post('/signup', function (req, res, next) {
+function postSignup(req, res, next) {
   passport.authenticate('local-signup', {
     successRedirect: '/',
     failureRedirect: '/signup',
     failureFlash: true })(req, res, next);
-});
+};
 
-router.get('/signin', function (req, res, next) {
-  res.render('auth/signin.ejs', { message: req.flash('signinMessage') });
-});
+function getLogin(req, res, next) {
+  res.render('auth/login.ejs', { message: req.flash('loginMessage') });
+};
 
-// SIGN-IN: Authenticate the user
-router.post("/signin", function (req, res, next) {
-  passport.authenticate('local-signin', {
-    successRedirect: '/',
-    failureRedirect: '/signin',
-    failureFlash: true })(req, res, next);
-});
+// POST /login
+function postLogin(req, res, next) {
+  passport.authenticate('local-login', {
+    successRedirect : '/',
+    failureRedirect : '/login',
+    failureFlash    : true })(req, res, next);
+};
 
-router.get("/signout", function (req, res, next){
+// GET /logout
+function getLogout(req, res, next) {
   req.logout();
-  res.redirect("/");
-});
+  res.redirect('/');
+};
+
+// GET /auth/facebook
+function getFacebook() {
+  passport.authenticate('facebook', { scope: 'email'} );
+};
 
-router.get('/auth/facebook', passport.authenticate('facebook', { scope: 'email'} ));
+// GET /auth/facebook/callback
+function getFacebookCallback() {
+  passport.authenticate('facebook', {
+    successRedirect: '/',
+    failureRedirect: '/'
+  });
+};
+
+module.exports = {
```

Comment:
Please not that you are exporting an object (hash), not a function. When we export controllers previously, we export them as a function that takes an app, that's why require doesn't work when you require it

```js
module.exports = function (app) {
  app.use('/', router);
};
```